src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -954,15 +997,35 @@ Subgraph::SubgraphExecutor::SubgraphExecutor(const std::shared_ptr<Subgraph::Sub
     OPENVINO_ASSERT(!ov::snippets::utils::is_dynamic_value(m_buffer_scratchpad_size),
                     "Undefined buffer scratchpad size!");
     m_internal_buffer_size = static_cast<size_t>(m_nthreads) * m_buffer_scratchpad_size;
-    m_in_requested_descs = snippet_config->m_in_requested_descs;
-    const auto external_repacking_buffer_size =
-        std::accumulate(m_in_requested_descs.begin(),
-                        m_in_requested_descs.end(),
+
+#ifdef OPENVINO_ARCH_X86_64
+    m_repacking_impl_type = snippet_config->repacking_impl_type;
+    m_repacked_inputs = snippet_config->repacked_inputs;
+
+    auto external_buffer_size =
+        std::accumulate(m_repacked_inputs.begin(),
+                        m_repacked_inputs.end(),
                         size_t(0),
-                        [](size_t sum, const std::pair<size_t, ov::intel_cpu::MemoryDescPtr>& requested_desc_elem) {
-                            return sum + requested_desc_elem.second->getCurrentMemSize();
+                        [](size_t sum, const std::pair<size_t, CPURuntimeConfig::RepackedInput>& p) {
+                            return sum + p.second.desc()->getCurrentMemSize();
                         });
-    m_buffer_scratchpad = allocator(m_internal_buffer_size + external_repacking_buffer_size);
+
+    if (should_repacking_be_in_parallel()) {
+        // When external repacking is applied in parallel section,
+        // each thread should have own buffer to store repacked data
+        external_buffer_size *= m_nthreads;
+
+        // To avoid extra overheads in runtime on unordered_map creation,
```

Comment:
`m_repacked_offsets_by_threads` is a vector, not unordered_map. Could you please correct the comment?